frontend/src/providers/AuthProvider.jsx

```diff
@@ -0,0 +1,36 @@
+import { createContext, useEffect, useState } from 'react';
+import Cookie from 'js-cookie';
+
+export const authContext = createContext();
+
+const AuthProvider = (props) => {
+  const [user, setUser] = useState('');
+
+  // Run once to check if user has cookie already.
+  useEffect(() => {
+    const foundCookie = Cookie.get('email');
+    if (foundCookie) {
+      setUser(foundCookie);
+    }
+  }, []);
+
+  const login = (email) => {
+    setUser(email);
+    Cookie.set('email', email);
+  };
+
+  const logout = () => {
+    setUser('');
+    Cookie.set('email', '');
+  };
+
+  const userData = { user, login, logout };
+
+  return (
+    <authContext.Provider value={userData}>
+      { props.children }
+    </authContext.Provider>
+  );
+};
+
+export default AuthProvider;
```

Comment:
For security reasons from CORS or something I guess, the client is not able to access our server given cookies ie. the `req.session.user` so instead this adds a client side cookie of just the email that was returned to our login call and uses that to check if user is logged in or not.